src/libraries/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/SymmetricAlgorithm.cs

```diff
@@ -222,6 +222,209 @@ public bool ValidKeySize(int bitLength)
+        ///   </para>
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <para>
+        ///   The padding mode <see cref="PaddingMode.None" /> was used, but <paramref name="plaintextLength" />
+        ///   is not a whole number of blocks.
+        ///   </para>
+        /// </exception>
+        public int GetCiphertextLengthEcb(int plaintextLength, PaddingMode paddingMode) =>
+            GetCiphertextLengthBlockAligned(plaintextLength, paddingMode);
+
+        /// <summary>
+        /// Gets the length of a ciphertext with a given padding mode and plaintext length in CBC mode.
+        /// </summary>
+        /// <param name="paddingMode">The padding mode used to pad the plaintext to the algorithm's block size.</param>
+        /// <param name="plaintextLength">The plaintext length, in bytes.</param>
+        /// <returns>The length, in bytes, of the ciphertext with padding.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <para>
+        ///   <paramref name="plaintextLength" /> is a negative number.
+        ///   </para>
+        ///   <para>
+        ///   - or -
+        ///   </para>
+        ///   <para>
+        ///   <paramref name="plaintextLength" /> when padded is too large to represent as
+        ///   a signed 32-bit integer.
+        ///   </para>
+        ///   <para>
+        ///   - or -
+        ///   </para>
+        ///   <para>
+        ///   <paramref name="paddingMode" /> is not a valid padding mode.
+        ///   </para>
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <para>
+        ///   <see cref="BlockSize" /> is not a positive integer.
+        ///   </para>
+        ///   <para>
+        ///   - or -
+        ///   </para>
+        ///   <para>
+        ///   <see cref="BlockSize" /> is not a whole number of bytes. It must be divisible by 8.
+        ///   </para>
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <para>
+        ///   The padding mode <see cref="PaddingMode.None" /> was used, but <paramref name="plaintextLength" />
+        ///   is not a whole number of blocks.
+        ///   </para>
+        /// </exception>
+        public int GetCiphertextLengthCbc(int plaintextLength, PaddingMode paddingMode = PaddingMode.PKCS7) =>
+            GetCiphertextLengthBlockAligned(plaintextLength, paddingMode);
+
+        private int GetCiphertextLengthBlockAligned(int plaintextLength, PaddingMode paddingMode)
+        {
+            if (plaintextLength < 0)
+                throw new ArgumentOutOfRangeException(nameof(plaintextLength), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            int blockSizeBits = BlockSize; // The BlockSize property is in bits.
+
+            if (blockSizeBits <= 0 || (blockSizeBits & 0b111) != 0)
+                throw new CryptographicException(SR.Cryptography_UnsupportedBlockSize);
```

Comment:
I'm a little fuzzy if this should be a `CryptographicException`. It's not really an argument, so it made the most sense.